cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -440,6 +461,12 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
           out_buffers.emplace_back(static_cast<size_type*>(out_buf.data()) + out_buf.size);
           final_offsets.emplace_back(static_cast<size_type>(col_string_sizes[idx]));
         }
+        // Nested large strings column
+        else if (input_col.nesting_depth() > 0) {
+          CUDF_EXPECTS(static_cast<int64_t>(h_initial_str_offsets[idx]) >= int64_t{0},
+                       "Initial string offset must be >= 0");
```

Comment:
I don't like that this check relies on the behavior of static_cast. Could we just check for size_t::max?